src/lib/i18n/index.es

```diff
@@ -0,0 +1,211 @@
+/**
+ * Default language.
+ * @type {string}
+ */
+const DEFAULT_LANGUAGE = 'en'
+
+/**
+ * Pattern to detect params in a string.
+ * @type {Regexp}
+ */
+let paramsPattern = /%(\d+)/g
+
+/**
+ * To store all the created instances.
+ * @type {Object}
+ */
+let instances = {}
+
+/**
+ * Current language. Shared by all the instances.
+ * @type {String}
+ */
+let language = DEFAULT_LANGUAGE
+
+/**
+ * Public API.
+ * @type {Object}
+ */
+class I18N {
+
+  /**
+   * Creates an i18n instance.
+   * If the ID is specified returns a singleton for each ID.
+   * If the ID is not specified returns a new instance.
+   * @param {string} id Instance ID. Optional.
+   */
+  constructor (id) {
+    let isValidId = typeof id === 'string'
+
+    if (!!isValidId && !!instances[id]) {
```

Comment:
Agree with Oscar about not using isValidId several times. It would be easier tu use it once and force an early return in case of !isValidId (actually, you described it as it was case 1 but, in the code, it's not).